astronomy/lunar_eclipse_test.cpp

```diff
@@ -238,86 +239,92 @@ Length LunarEclipseTest::atmospheric_depth_;
-
-  CheckLunarPenumbralEclipse(P1, U14, 2e-5 * Radian, 28 * Second);
-  CheckLunarUmbralEclipse(U1, U14,    2e-5 * Radian, 30 * Second);
-  CheckLunarUmbralEclipse(U2, U23,    2e-5 * Radian, 39 * Second);
-  CheckLunarUmbralEclipse(U3, U23,    9e-6 * Radian, 21 * Second);
-  CheckLunarUmbralEclipse(U4, U14,    2e-5 * Radian, 28 * Second);
-  CheckLunarPenumbralEclipse(P4, U14, 2e-5 * Radian, 30 * Second);
-
-  // Times are TDB Julian Day for 1950-09-26.
-  P1 = JulianDate(2433550.55712016);  // 01:21:43 UT
-  U1 = JulianDate(2433550.60578913);  // 02:31:48
-  U2 = JulianDate(2433550.66325441);  // 03:54:33
-  U3 = JulianDate(2433550.69399515);  // 04:38:49
-  U4 = JulianDate(2433550.75144885);  // 06:01:33
-  P4 = JulianDate(2433550.800222);    // 07:11:47
-
-  CheckLunarPenumbralEclipse(P1, U14, 2e-5 * Radian, 34 * Second);
-  CheckLunarUmbralEclipse(U1, U14,    3e-5 * Radian, 36 * Second);
-  CheckLunarUmbralEclipse(U2, U23,    3e-5 * Radian, 42 * Second);
-  CheckLunarUmbralEclipse(U3, U23,    2e-5 * Radian, 29 * Second);
-  CheckLunarUmbralEclipse(U4, U14,    3e-5 * Radian, 34 * Second);
-  CheckLunarPenumbralEclipse(P4, U14, 2e-5 * Radian, 36 * Second);
+  LOG(ERROR)<<"1950-04-02T20:48:47.5"_UT1-"1950-04-02T20:49:16.7"_TT;
+  {
+    constexpr auto P1 = "1950-04-02T18:10:49"_UT1;
+    constexpr auto U1 = "1950-04-02T19:09:19"_UT1;
+    constexpr auto U2 = "1950-04-02T20:30:38"_UT1;
+    constexpr auto U3 = "1950-04-02T20:57:33"_UT1;
+    constexpr auto U4 = "1950-04-02T22:18:54"_UT1;
+    constexpr auto P4 = "1950-04-02T23:17:21"_UT1;
+
+    CheckLunarPenumbralEclipse(P1, U14, 2e-5 * Radian, 31 * Second);
+    CheckLunarUmbralEclipse(U1, U14, 3e-5 * Radian, 33 * Second);
+    CheckLunarUmbralEclipse(U2, U23, 2e-5 * Radian, 42 * Second);
+    CheckLunarUmbralEclipse(U3, U23, 1e-5 * Radian, 24 * Second);
+    CheckLunarUmbralEclipse(U4, U14, 2e-5 * Radian, 31 * Second);
+    CheckLunarPenumbralEclipse(P4, U14, 2e-5 * Radian, 33 * Second);
+  }
+
+  {
+    constexpr auto P1 = "1950-09-26T01:21:43"_UT1;
+    constexpr auto U1 = "1950-09-26T02:31:48"_UT1;
+    constexpr auto U2 = "1950-09-26T03:54:33"_UT1;
+    constexpr auto U3 = "1950-09-26T04:38:49"_UT1;
+    constexpr auto U4 = "1950-09-26T06:01:33"_UT1;
+    constexpr auto P4 = "1950-09-26T07:11:47"_UT1;
+
+    CheckLunarPenumbralEclipse(P1, U14, 2e-5 * Radian, 37 * Second);
+    CheckLunarUmbralEclipse(U1, U14, 3e-5 * Radian, 39 * Second);
+    CheckLunarUmbralEclipse(U2, U23, 3e-5 * Radian, 45 * Second);
+    CheckLunarUmbralEclipse(U3, U23, 2e-5 * Radian, 32 * Second);
+    CheckLunarUmbralEclipse(U4, U14, 3e-5 * Radian, 37 * Second);
+    CheckLunarPenumbralEclipse(P4, U14, 2e-5 * Radian, 38 * Second);
+  }
 }
 
 TEST_F(LunarEclipseTest, Year1951) {
-  // Times are TDB Julian Day for 1951-03-23.
-  auto P1 = JulianDate(2433728.86842806);  // 08:50:50
-  auto P4 = JulianDate(2433729.01725909);  // 12:24:19
+  {
+    constexpr auto P1 = "1951-03-23T08:50:50"_UT1;
```

Comment:
`08:50:50` was a typo by @pdn4kd (note the anomalous error). Should be `08:50:00`. http://eclipse.gsfc.nasa.gov/LEplot/LEplot1951/LE1951Mar23N.pdf